graylog2-web-interface/src/views/components/searchbar/QueryInput.tsx

```diff
@@ -62,23 +70,32 @@ const handleExecution = (editor, onExecute, value, error) => {
   onExecute(value);
 };
 
-const _configureEditor = (node, completer, onExecute) => {
-  const editor = node && node.editor;
+const displaySuggestionAfterFieldName = (editor, currentLine: number, lines: Array<Array<Line>>) => {
+  const currentToken = lines[currentLine - 1].find((token) => token?.start !== undefined);
 
-  if (editor) {
-    editor.commands.addCommands([{
-      name: 'Execute',
-      bindKey: 'Enter',
-      exec: onExecute,
-    },
-    {
-      name: 'SuppressShiftEnter',
-      bindKey: 'Shift-Enter',
-      exec: () => {},
-    }]);
+  if (!currentToken) {
+    return;
+  }
 
+  if (currentToken.type === 'keyword' && currentToken.value.endsWith(':')) {
+    editor.execCommand('startAutocomplete');
+  }
+};
+
+const _configureEditor = (editor, completer: AutoCompleter, configuredListeners: React.MutableRefObject<boolean>) => {
+  if (editor) {
     editor.commands.removeCommands(['find', 'indent', 'outdent']);
+    // eslint-disable-next-line no-param-reassign
     editor.completers = [completer];
+
+    if (!configuredListeners.current) {
+      editor.session.on('tokenizerUpdate', (input, { bgTokenizer: { currentLine, lines } }) => {
+        displaySuggestionAfterFieldName(editor, currentLine, lines);
```

Comment:
In a previous version, tried a different approach to display the suggestions, mainly by adjusting the regex which identifies the word we want to display completions for and by including the field name in the field value suggestions.
This had some disadvantages:
- it only worked well, when there were no other field name suggestions. When typing in `test-field:` while there are still suggestion like `test-field-1`, the field value suggestions are not being displayed.
- the field names were considered when filtering the suggestions. Imagine there are four field value suggestions when typing in "test-field:". If you start typing "test-field:tes" it will still display all other suggestions, because the prefix of the value is part of the field name.

With the current approach we are opening the suggestions manually, it has the disadvantage, that the `QueryInput` contains code which is related to the field name suggestions related code and that we are not relying on the ace editor logic On the other hand it has the advantage, that we could now refetch the suggestions while typing in the field name.